### PR TITLE
Don't rely on the "moola" petname; use the brand registry key

### PR DESCRIPTION
**Pull request by michaelfig**
Generalise support not to allow specifying `TIP_ISSUER_NAME=simoleans agoric deploy api/deploy.js` to specify "simoleans" instead of "moola".  Then fetch the deployer's wallet's issuer under that petname and use it with its brandRegKey.

This is because people's petnames for their purses or issuers are supposed to be able to vary per person.  In the future, it would be great to allow people to install any arbitrary issuer on the publicAPI before using it for tipping.

### CODE REVIEW

**Comment by katelynsills** [comment]
I like the idea of not hardcoding the subsequent steps to be for moola, but I don't like the idea of fetching the wallets first issuer. If a user was using this for real, they would have a particular issuer in mind.

File: api/deploy.js [hunk 1]
```diff
@@ -94,14 +94,20 @@ export default async function deployApi(referencesPromise, { bundleSource, pathR
   // registered the moolaIssuer in the registry. We could also get it
   // from our wallet.
 
+  // get
```
Comment (katelynsills) [inline api/deploy.js:97]: Is this accidental?

**Comment by michaelfig** [comment]
> I like the idea of not hardcoding the subsequent steps to be for moola, but I don't like the idea of fetching the wallet's first issuer. If a user was using this for real, they would have a particular issuer in mind.

Ah, okay, so the user should be able to determine which of their wallet's issuers to use?  That makes sense.

I'll take another attempt at this.

File: api/deploy.js [hunk 2]
```diff
@@ -94,14 +94,20 @@ export default async function deployApi(referencesPromise, { bundleSource, pathR
...
   // getIssuers returns an array, because we currently cannot
   // serialize maps. We can immediately create a map using the array,
   // though. https://github.com/Agoric/agoric-sdk/issues/838
   const issuersArray = await E(wallet).getIssuers();
-  const issuers = new Map(issuersArray);
-  const moolaIssuer = issuers.get('moola');
+  
+  // Pick the first issuer in the wallet as our tip issuer.
+  const tipIssuer = issuersArray[0][1];
```
Comment (katelynsills) [inline api/deploy.js:104]: I'd rather not pick an issuer at random. I think the user would have an issuer in mind that they would select through petname from their wallet. I think we should revert this change.

File: api/deploy.js [hunk 3]
```diff
@@ -94,14 +94,20 @@ export default async function deployApi(referencesPromise, { bundleSource, pathR
...
+
+  // Find its brand registry key so we can communicate the issuer to other wallets.
+  // Equivalent to: await wallet~.getIssuerNames(tipIssuer)~.brandRegKey
+  const TIP_BRAND_REGKEY = await E.G(E(wallet).getIssuerNames(tipIssuer)).brandRegKey;
```
Comment (katelynsills) [inline api/deploy.js:108]: I think everything below this were we don't hardcode the issuer sounds great! Thank you for doing that.

File: api/deploy.js (hunk 2 above)
Comment (michaelfig) [reply api/deploy.js]: Done, in favour of a configurable environment variable.

File: api/deploy.js (hunk 1 above)
Comment (michaelfig) [reply api/deploy.js]: Yep, removed.
